### ISSUE TITLE
Parsing of entities in multipart/alternative has regressed between 0.4.3 and 0.5

### ISSUE THREAD

**Issue by romanofski** [purebred-mua/purebred-email#78]
I noticed that Purebred all of a sudden isn't parsing entities correctly any more on certain mails. I did a bisect today and this commit:

```
commit cbef915e1e7f31631e7a604de8bcf53e6e57a3cb
Author: Fraser Tweedale <frase@frase.id.au>
Date:   Fri Oct 1 10:30:57 2021 +1000

    add function for general multipart Content-Type creation
```
introduced the regression.

I don't know yet what's causing it, but it seems the mail's main body has structured like:
```
MIME-Version: 1.0
Date: Thu, 4 May 2017 03:08:43 +1000 (EST)
From: foo@bar
To: my@email.test
Subject: Pretty subject
Content-Type: multipart/related; boundary="boundary1"

--boundary1
Content-Type: multipart/alternative; boundary="boundary2"

--boundary2
Content-Type: text/plain; charset="UTF-8"

--boundary2
Content-Type: text/html; charset="UTF-8"

Bla bla bla
--boundary2--
--boundary1--
```
I hacked up the parse tool and did a bisect to find the regressing commit. Don't know yet what's tripping up the parser.

**Comment by romanofski** [comment]
Update: Symptom is, that `toListOf entities msg` where `msg` is the parsed message returns an empty list.

**Comment by frasertweedale** [comment]
@romanofski duplicate of https://github.com/purebred-mua/purebred-email/issues/68, which was fixed in 73be2d7bbc1d012566fee826b4dd1c863a897732 but remains unreleased.
